applications/h264/grpc_h264_endoscopy_tool_tracking/cpp/edge/video_input_fragment.hpp

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights
+ * reserved. SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_EDGE_VIDEO_INPUT_FRAGMENT_HPP
+#define GRPC_H264_ENDOSCOPY_TOOL_TRACKING_CPP_EDGE_VIDEO_INPUT_FRAGMENT_HPP
+
+#include <holoscan/holoscan.hpp>
+#include <holoscan/operators/format_converter/format_converter.hpp>
+#include <holoscan/operators/gxf_codelet/gxf_codelet.hpp>
+#include <holoscan/operators/holoviz/holoviz.hpp>
+
+#include <grpc_client.hpp>
+
+#include "../gxf_imports.hpp"
+#include "video_input_fragment.hpp"
+#include "viz_fragment.hpp"
+
+namespace holohub::grpc_h264_endoscopy_tool_tracking {
+using namespace holoscan;
+using namespace holoscan::ops;
+
+/**
+ * @class VideoInputFragment
+ * @brief A fragment class for handling video input and processing.
+ *
+ * This class is responsible for reading video bitstreams, sending requests to a gRPC client for
+ * inference, and processing. The video bitstreams also decodes video frames for the Holoviz to
+ * display the frames.
+ */
+class VideoInputFragment : public holoscan::Fragment {
+ public:
+  explicit VideoInputFragment(const std::string& datapath, const uint32_t width,
+                              const uint32_t height)
+      : datapath_(datapath), width_(width), height_(height) {}
+
+  ~VideoInputFragment() { entity_client_service_->stop_entity_stream(); }
+
+  void compose() override {
+    condition_ = make_condition<AsynchronousCondition>("response_available_condition");
+    request_queue_ =
+        make_resource<ConditionVariableQueue<std::shared_ptr<EntityRequest>>>("request_queue");
+    response_queue_ =
+        make_resource<AsynchronousConditionQueue<std::shared_ptr<nvidia::gxf::Entity>>>(
+            "response_queue", condition_);
+
+    auto bitstream_reader = make_operator<VideoReadBitstreamOp>(
+        "bitstream_reader",
+        from_config("bitstream_reader"),
+        Arg("input_file_path", datapath_ + "/surgical_video.264"),
+        make_condition<CountCondition>(750),
+        make_condition<PeriodicCondition>("periodic-condition",
+                                          Arg("recess_period") = std::string("25hz")),
+        Arg("pool") = make_resource<UnboundedAllocator>("pool"));
```

Comment:
Also here, should replace UnboundAllocator with RMMAllocator and add CudaStreamPool to operators which support it.

Reply:
I have replaced `UnboundedAllocator` with `RMMAllocator`.